crates/api_crud/src/user/delete.rs

```diff
@@ -19,8 +19,29 @@ use lemmy_utils::error::{LemmyErrorType, LemmyResult};
 pub async fn delete_account(
   data: Json<DeleteAccount>,
   context: Data<LemmyContext>,
-  local_user_view: LocalUserView,
+  local_user_view_opt: Option<LocalUserView>,
 ) -> LemmyResult<Json<SuccessResponse>> {
+  // If a local_user_view exists, that means they're logged in.
+  let local_user_view = if let Some(local_user_view) = local_user_view_opt {
+    local_user_view
+  } else if let Some(username_or_email) = &data.username_or_email {
+    // Otherwise, they're likely banned, meaning we need to validate their login.
+    let local_user_view =
+      LocalUserView::find_by_email_or_name(&mut context.pool(), username_or_email).await?;
+
+    // Only check TOTP if they're not logged in.
+    if local_user_view.local_user.totp_2fa_enabled {
+      check_totp_2fa_valid(
+        &local_user_view,
+        &data.totp_2fa_token,
+        &context.settings().hostname,
+      )?;
+    }
```

Comment:
it would probably be better to move the password validation into this section to keep the check order the same as for logins, where first the password is verified and only then the totp is verified.

ideally password and totp would be be validated in same step and return the same error regardless of which one is incorrect, but that's out of scope for this PR.

I also noticed that this endpoint currently only has the message rate limit if i read it correctly; if this is now performing authentication it should probably get the register rate limit added as the login endpoint has.

Reply:
Done